tool/checkstyle/rules.bzl

```diff
@@ -67,33 +64,39 @@ def _checkstyle_test_impl(ctx):
     )
 
     ctx.actions.expand_template(
-        template = ctx.file._checkstyle_py_template,
-        output = ctx.outputs.checkstyle_script,
+        template = ctx.file._checkstyle_kt_template,
+        output = ctx.outputs.checkstyle_wrapper,
         substitutions = {
             "{command}" : cmd,
-            "{allow_failure}": str(int(ctx.attr.allow_failure)),
         },
         is_executable = True,
     )
 
-    files = [ctx.outputs.checkstyle_script] + ctx.files._license_files + files + ctx.files._classpath + inputs
+    files = [ctx.outputs.checkstyle_wrapper] + ctx.files._license_files + files + ctx.files._classpath + inputs
     runfiles = ctx.runfiles(
         files = files,
         collect_data = True,
     )
     return DefaultInfo(
-        executable = ctx.outputs.checkstyle_script,
+        executable = ctx.outputs.checkstyle_wrapper,
         files = depset(files),
         runfiles = runfiles,
     )
 
-checkstyle_test = rule(
+_checkstyle_test = rule(
```

Comment:
Another layer of indirection. This rule will later be wrapped in a `kt_jvm_test()` in the real `checkstyle_test()`.